### PR TITLE
[Docs] Remove `ELLIPSIS` doctest option flag

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The `ELLIPSIS` doctest option flag is enabled by default, so we don't need to manually set it in docstring examples.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-contribute/writing-code-snippets.rst [hunk 1]
```diff
@@ -221,24 +221,10 @@ If your Python code is non-deterministic, or if your output is excessively long,
...
-        >>> import ray
-        >>> ray.data.read_images("example://image-datasets/simple")
-        Dataset(
-           num_blocks=...,
-           num_rows=...,
-           schema={image: numpy.ndarray(shape=(32, 32, 3), dtype=uint8)}
-        )
-
-If you omit the `doctest` directive, append `# doctest: +ELLIPSIS` to your code instead.
+To ignore parts of a *doctest-style* output, replace problematic sections with ellipsis. ::
```
Comment (jjyao) [inline doc/source/ray-contribute/writing-code-snippets.rst:224]: What should I do if I want to exact match `...`?

Comment (bveeramani) [reply doc/source/ray-contribute/writing-code-snippets.rst:224]: You mean to literally match "..." and not match everything? You can disable it with `:options: -ELLIPSIS`, I think.

```
.. testcode::

    print("...")

.. testoutput::
    :options: -ELLIPSIS
    ...
```

Comment (jjyao) [reply doc/source/ray-contribute/writing-code-snippets.rst:224]: Yea, not sure whether people will ever want this but probably worth mentioning in the `writing-code-snippets.rst`

File: doc/source/ray-contribute/writing-code-snippets.rst [hunk 2]
```diff
@@ -250,8 +236,8 @@ To ignore an output altogether, write a *code-block-style* snippet. Don't use `#
 Ignoring *code-block-style* outputs
 ===================================
 
-If parts of your output are long or non-deterministic, add `:options: +ELLIPSIS` to
-the `testoutput` directive and replace problematic sections with ellipsis. ::
+If parts of your output are long or non-deterministic, replace problematic sections
+with ellipsis. ::
```
Comment (angelinalg) [inline doc/source/ray-contribute/writing-code-snippets.rst:240]:
```suggestion
with ellipses. ::
```

File: doc/source/ray-contribute/writing-code-snippets.rst [hunk 3]
```diff
@@ -281,16 +266,15 @@ If your output is nondeterministic and you want to display a sample output, add
 
         0.969461416250246
 
-If your output is hard to test and you don't want to display a sample output, add
-`:options: +ELLIPSIS` and `:hide:`. ::
+If your output is hard to test and you don't want to display a sample output, use
+ellipsis and `:hide:`. ::
```
Comment (angelinalg) [inline doc/source/ray-contribute/writing-code-snippets.rst:270]:
```suggestion
ellipses and `:hide:`. ::
```

**Comment by angelinalg** [comment]
`ellipsis` vs `ellipses` is messing with my head a bit. Your sentence structures lead me think you want to use `ellipses`, like you would say, "use parentheses", or "use quotes". If you mean to use the singular, then consider using "an ellipsis".

**Review by scottjlee — APPROVED** [review]

**Review by angelinalg — APPROVED** [review]

**Comment by bveeramani** [comment]
> `ellipsis` vs `ellipses` is messing with my head a bit. Your sentence structures lead me think you want to use `ellipses`, like you would say, "use parentheses", or "use quotes". If you mean to use the singular, then consider using "an ellipsis".

Ah yeah, you're right. Didn't realize there was a distinction .

File: doc/source/ray-contribute/writing-code-snippets.rst (hunk 1 above)
Comment (matthewdeng) [reply doc/source/ray-contribute/writing-code-snippets.rst]: @bveeramani is there is a decision for this?

Comment (bveeramani) [reply doc/source/ray-contribute/writing-code-snippets.rst]: I don't think people will ever want to do this, so I'm not sure if it's worth documenting it. I want to keep the guide simple.

If people run into this issue later, I'll document this.

**Pull request merged**